message-dispatcher/message-dispatcher.bal

```diff
@@ -0,0 +1,21 @@
+import ballerina/http;
+
+final http:Client ocrClient = check new ("http://api.ocr.com.balmock.io");
+
+type OcrResponse record {|
+    string[] lines;
+    int noOfLines;
+    string pdfUrl;
+|};
+
+service / on new http:Listener(8080) {
+    string[] ocrProcessors = ["processor1", "processor2", "processor3"];
+    int processorNo = 0;
+
+    resource function get ocr(string url) returns OcrResponse|error {
```

Comment:
this scenario is related to acquiring the processor resources. So we should use make this function concurrently safe.

Reply:
If we can have a locking mechanism for each item in the array, it would make more sense.